components/style/values/computed/font.rs

```diff
@@ -70,6 +81,69 @@ impl From<specified::KeywordSize> for KeywordInfo {
     }
 }
 
+impl FontWeight {
+    /// Value for normal
+    pub fn normal() -> Self {
+        FontWeight(400)
+    }
+
+    /// Value for bold
+    pub fn bold() -> Self {
+        FontWeight(700)
+    }
+
+    /// Convert from an integer to Weight
+    pub fn from_int(n: i32) -> Result<Self, ()> {
+        if n >= 100 && n <= 900 && n % 100 == 0 {
+            Ok(FontWeight(n as u16))
+        } else {
+            Err(())
+        }
+    }
+
+    /// Convert from an Gecko weight
+    pub fn from_gecko_weight(weight: u16) -> Self {
+        // we allow a wider range of weights than is parseable
+        // because system fonts may provide custom values
+        FontWeight(weight)
+    }
+
+    /// Weither this weight is bold
+    pub fn is_bold(&self) -> bool {
+        self.0 > 500
+    }
+
+    /// Return the bolder weight
+    pub fn bolder(self) -> Self {
+        if self.0 < 400 {
+            FontWeight(400)
+        } else if self.0 < 600 {
+            FontWeight(700)
+        } else {
+            FontWeight(900)
+        }
+    }
+
+    /// Returns the lighter weight
+    pub fn lighter(self) -> Self {
+        if self.0 < 600 {
+            FontWeight(100)
+        } else if self.0 < 800 {
+            FontWeight(400)
+        } else {
+            FontWeight(700)
+        }
+    }
+}
+
+impl Parse for FontWeight {
```

Comment:
This is pre-existing, but it feels weird to use `parse` in a `computed` type, and this can be more efficient if implemented in the `specified` type directly.